internal/api/httpsrv/handler.go

```diff
@@ -0,0 +1,127 @@
+package httpsrv
+
+import (
+	"context"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+
+	"go.infratographer.com/identity-manager-sts/internal/storage"
+	"go.infratographer.com/identity-manager-sts/internal/types"
+	v1 "go.infratographer.com/identity-manager-sts/pkg/api/v1"
+)
+
+// apiHandler represents an API handler.
+type apiHandler struct {
+	engine storage.Engine
+}
+
+func (h *apiHandler) CreateIssuer(ctx context.Context, req CreateIssuerRequestObject) (CreateIssuerResponseObject, error) {
+	createOp := req.Body
+
+	var (
+		claimsMapping types.ClaimsMapping
+		err           error
+	)
+
+	if createOp.ClaimMappings != nil {
+		claimsMapping, err = types.NewClaimsMapping(*createOp.ClaimMappings)
+		if err != nil {
+			return CreateIssuer400JSONResponse{
+				Errors: []string{
+					"error parsing CEL expression",
+				},
+			}, nil
+		}
+	}
+
+	issuerToCreate := types.Issuer{
+		ID:            uuid.New().String(),
+		Name:          createOp.Name,
+		URI:           createOp.URI,
+		JWKSURI:       createOp.JWKSURI,
+		ClaimMappings: claimsMapping,
+	}
+
+	issuer, err := h.engine.Create(ctx, issuerToCreate)
+	if err != nil {
+		return nil, err
+	}
+
+	claimsMappingRepr, err := issuer.ClaimMappings.Repr()
+	if err != nil {
+		return nil, err
+	}
+
+	out := v1.Issuer{
+		ID:            uuid.MustParse(issuer.ID),
+		Name:          issuer.Name,
+		URI:           issuer.URI,
+		JWKSURI:       issuer.JWKSURI,
+		ClaimMappings: claimsMappingRepr,
+	}
+
+	return CreateIssuer200JSONResponse(out), nil
+}
+
+func (h *apiHandler) GetIssuerByID(ctx context.Context, req GetIssuerByIDRequestObject) (GetIssuerByIDResponseObject, error) {
+	id := req.Id.String()
+
+	iss, err := h.engine.GetByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	claimMappingRepr, err := iss.ClaimMappings.Repr()
```

Comment:
`iss` pointer should be checked for nil value.

Reply:
To an extent we need to trust that the return signatures for functions make sense. By convention in Go, if a function returns `(*T, error)` and the error is nil, the pointer that's returned should be non-nil. If it isn't, I would argue that the better response is actually to panic rather than try to gracefully fail because the function did something that doesn't make sense.